### PR TITLE
Fixes for coverity scan (part 2)

### PR DESCRIPTION
**Pull request by dgarske**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -6944,16 +6971,14 @@ static int DoCertificate(WOLFSSL* ssl, byte* input, word32* inOutIdx,
...
-                                args->dCert->sigCtx.asyncDev,
-                                WC_ASYNC_FLAG_CALL_AGAIN);
-                        }
-                    #endif
+                #ifdef WOLFSSL_ASYNC_CRYPT
+                    if (ret == WC_PENDING_E) {
+                        ret = wolfSSL_AsyncPush(ssl,
+                            args->dCert->sigCtx.asyncDev,
+                            WC_ASYNC_FLAG_CALL_AGAIN);
                         goto exit_dc;
```
Comment (toddouska) [inline src/internal.c]: This goto exit_dc is now only in ASYNC build. That doesn't seem right.  Where is the ParseCertRelative failure handled in non async or an async failure that isn't WC_PENDING_E?

File: src/internal.c [hunk 2]
```diff
@@ -7110,16 +7135,14 @@ static int DoCertificate(WOLFSSL* ssl, byte* input, word32* inOutIdx,
...
```
Comment (toddouska) [inline src/internal.c]: Same as previous comment.

File: src/internal.c (hunk 1 above)
Comment (dgarske) [inline src/internal.c]: The Coverity report indicated some dead code below with ret checking and when I looked back to pre-async the behavior was to use the ret checking code below. So only the WC_PENDING_E case needs to do a exit_dc. This fix restores pre-async error handling for ParseCertRelative in this function.

File: src/internal.c [hunk 3]
```diff
@@ -5330,8 +5330,10 @@ static INLINE int GrowOutputBuffer(WOLFSSL* ssl, int size)
 #if WOLFSSL_GENERAL_ALIGNMENT > 0
     byte  hdrSz = ssl->options.dtls ? DTLS_RECORD_HEADER_SZ :
                                       RECORD_HEADER_SZ;
-#endif
+    byte align = WOLFSSL_GENERAL_ALIGNMENT;
+#else
     const byte align = WOLFSSL_GENERAL_ALIGNMENT;
```
Comment (toddouska) [inline src/internal.c]: commit title says to only use align when WOLFSSL_GENERAL_ALIGNMENT is > 0.  But align here is set to WOLFSSL_GENERAL_ALIGNMENT and then used as an offset in XMALLOC.  In the zero case this is fine, in the negative case (which I'm guessing is the issue), not so good.

Comment (dgarske) [inline src/internal.c]: The scan only found a dead code scenario where WOLFSSL_GENERAL_ALIGNMENT is zero. This was to "cleanup" the function for the case when no align is defined (0). I agree that WOLFSSL_GENERAL_ALIGNMENT should never be a negative value. We could add an #if compile - time check for that in a later commit.

**Pull request merged**